Lib/sqlite3/test/dbapi.py

```diff
@@ -933,6 +935,72 @@ def test_on_conflict_replace(self):
         self.assertEqual(self.cu.fetchall(), [('Very different data!', 'foo')])
 
 
+class MultiprocessTests(unittest.TestCase):
+    def setUp(self):
+        with sqlite.connect(TESTFN) as cx:
+            cx.execute("create table t(t)")
+
+    def tearDown(self):
+        unlink(TESTFN)
+
+    def test_rollback_if_db_is_locked(self):
+        # bpo-27334: ctx manager does not rollback if commit fails
+        CONNECTION_TIMEOUT = SHORT_TIMEOUT / 1000.  # Defaults to 30 ms
+        SCRIPT = f"""if 1:
+            import sqlite3
+            input()  # wait for parent process to start
+            cx = sqlite3.connect("{TESTFN}", timeout={CONNECTION_TIMEOUT})
+            try:
+                with cx:
+                    cx.execute("insert into t values('test')")
+            except Exception as exc:
+                print(exc)
+            else:
+                print("no error in sub")
+            input()  # keep proc open till parent calls
+        """
+
+        # spawn child process
+        proc = subprocess.Popen(
+            [sys.executable, "-c", SCRIPT],
+            encoding="utf-8",
+            bufsize=0,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        self.addCleanup(proc.communicate)
+
+        # connect to db, and create a UDF that waits for child
+        cx = sqlite.connect(TESTFN, timeout=CONNECTION_TIMEOUT)
+        def wait():
+            proc.stdin.write("\n")  # tell child to connect
```

Comment:
Hum, I don't know if `\n` works on Windows and macOS. Maybe use print(file=...)?

Reply:
Good point! Thanks.